code/game/objects/effects/decals/turfdecal/tilecoloring.dm

```diff
@@ -0,0 +1,152 @@
+/obj/effect/turf_decal/corner
+	icon_state = "corner_white"
+
+/obj/effect/turf_decal/corner/black
+	name = "black corner"
+	color = COLOR_ALMOST_BLACK
+
+/obj/effect/turf_decal/corner/black/diagonal
+	icon_state = "corner_white_diagonal"
+
+/obj/effect/turf_decal/corner/black/full
+	icon_state = "corner_white_full"
```

Comment:
These aren't actually full and should be part of the corner sprite instead as the intermediate directions.